src/plugins/drawing-tool/components/drawing-layer.tsx

```diff
@@ -817,33 +368,30 @@ export class DrawingLayerView extends React.Component<DrawingLayerViewProps, Dra
   }
 
   // when we add text, this filter can be used with this.renderObjects((object) => object.type !== "text")
-  public renderObjects(_filter: (object: DrawingObject) => boolean) {
+  public renderObjects(_filter: (object: DrawingObjectType) => boolean) {
     return Object.keys(this.objects).map((id) => {
       const object = this.objects[id];
       if (!object || !_filter(object)) {
         return null;
       }
-      return object.render({
-        id,
-        handleHover: this.handleObjectHover,
-        drawingLayer: this
-      });
+      return renderDrawingObject(object, this.getContent(), this.handleObjectHover);
     });
   }
 
-  public renderSelectedObjects(selectedObjects: DrawingObject[], color: string) {
+  public renderSelectedObjects(selectedObjects: DrawingObjectType[], color: string) {
     return selectedObjects.map((object, index) => {
-      const {nw, se} = object.getBoundingBox();
-      nw.x -= SELECTION_BOX_PADDING;
-      nw.y -= SELECTION_BOX_PADDING;
-      se.x += SELECTION_BOX_PADDING;
-      se.y += SELECTION_BOX_PADDING;
+      let {nw: {x: nwX, y: nwY}, se: {x: seX, y: seY}} = object.boundingBox;
+      nwX -= SELECTION_BOX_PADDING;
```

Comment:
Now that the boundBox is not created new each time, we cannot just update its values in place. Otherwise each call to renderSelectedObjects changes the boundBox.